dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyDataControllerTest.kt

```diff
@@ -402,4 +420,182 @@ class CompanyDataControllerTest {
+        )
+    }
+
+    @Test
+    fun `check that the a company owner can patch contact details and website of their own company`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val originalCompany = uploadInfo.actualStoredCompany
+        val companyId = originalCompany.companyId
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        apiAccessor.companyRolesControllerApi.assignCompanyRole(
+            CompanyRole.CompanyOwner,
+            UUID.fromString(companyId),
+            dataReaderUserId,
+        )
+
+        val patchObject = CompanyInformationPatch(
+            website = "New-Website-3",
+            companyContactDetails = setOf("New-Email-1@example.com"),
+        )
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+        val updatedCompany = apiAccessor.companyDataControllerApi.patchCompanyById(
+            companyId,
+            patchObject,
+        )
+        assertEquals(
+            patchObject.companyContactDetails!!, updatedCompany.companyInformation.companyContactDetails,
+            "The company contact details should have been updated",
+        )
+        assertEquals(
+            patchObject.website!!, updatedCompany.companyInformation.website,
+            "The company name should have been updated by the company owner",
+        )
+    }
+
+    @Test
+    fun `check that the data owner cannot patch unallowed fields`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val originalCompany = uploadInfo.actualStoredCompany
+        val companyId = originalCompany.companyId
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        apiAccessor.companyRolesControllerApi.assignCompanyRole(
+            CompanyRole.CompanyOwner,
+            UUID.fromString(companyId),
+            dataReaderUserId,
+        )
+
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+
+        assertThrows<ClientException> {
+            apiAccessor.companyDataControllerApi.patchCompanyById(
+                companyId,
+                fullPatchObject,
+            )
+        }
+    }
+
+    @Test
+    fun `check that patching with an invalid company ID and invalid patch fields returns an exception`() {
+        val uploadInfo = apiAccessor.uploadNCompaniesWithoutIdentifiers(1).first()
+        val originalCompany = uploadInfo.actualStoredCompany
+        val companyId = originalCompany.companyId
+
+        apiAccessor.jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+
+        assertThrows<ClientException> {
+            apiAccessor.companyDataControllerApi.patchCompanyById(
+                companyId,
+                fullPatchObject,
+            )
+        }
+    }
```

Comment:
Wieso steht in der Testbeschreibung, dass die companyId invalid ist?
Die companyId im Test ist doch valid.

Und was ist genau dein Ziel bei dem Test?
Möchtest du checken, dass eine invalid companyId zu einer "invalid company" error messsage führt?

Reply:
there was an alteration to the company id which went lost in the commits. re added it